lib/oli/delivery/attempts/activity_lifecycle/evaluate.ex

```diff
@@ -13,6 +13,87 @@ defmodule Oli.Delivery.Attempts.ActivityLifecycle.Evaluate do
   import Oli.Delivery.Attempts.Core
   import Oli.Delivery.Attempts.ActivityLifecycle.Persistence
 
+  require Logger
+
+  def evaluate_activity(section_slug, activity_attempt_guid, part_inputs) do
+    %ActivityAttempt{
+      transformed_model: transformed_model,
+      attempt_number: attempt_number,
+      resource_attempt: resource_attempt
+    } =
+      get_activity_attempt_by(attempt_guid: activity_attempt_guid)
+      |> Repo.preload([:resource_attempt])
+
+    {:ok, %Model{rules: rules}} = Model.parse(transformed_model)
+
+    if is_list(rules) do
+      evaluate_from_rules(section_slug, activity_attempt_guid, part_inputs)
+    else
+      evaluate_from_input(section_slug, activity_attempt_guid, part_inputs)
+    end
+  end
+
+  def evaluate_from_rules(section_slug, activity_attempt_guid, part_inputs) do
+    %ActivityAttempt{
+      transformed_model: transformed_model,
+      attempt_number: attempt_number,
+      resource_attempt: resource_attempt
+    } =
+      get_activity_attempt_by(attempt_guid: activity_attempt_guid)
+      |> Repo.preload([:resource_attempt])
+
+    {:ok, %Model{rules: rules}} = Model.parse(transformed_model)
+
+    # need to get all of the extrinsic (resource_attempt state)
+    # need to get *all* of the activity attempts state (part responses saved thus far)
+    attempt_hierarchy = Oli.Delivery.Attempts.PageLifecycle.Hierarchy.get_latest_attempts(resource_attempt.id)
+
+    response_state = Enum.reduce(Map.values(attempt_hierarchy), %{}, fn {activity_attempt, part_attempts}, m ->
+      part_responses = Enum.reduce(Map.values(part_attempts), %{}, fn pa, acc ->
+        case pa.response do
+          nil -> acc
+          _ ->
+            part_values = Enum.reduce(Map.values(pa.response), %{}, fn pv, acc1 ->
+              Map.put(acc1, Map.get(pv, "path"), Map.get(pv, "value"))
+            end)
```

Comment:
because the back end doesn't know about the "deck" layout or the sequence_id and the like, only the attempt_guid and response for it, the path is determined by the client that does when it makes a call to save the response.
the path looks like `path: "q:1555699921528:664|stage.Name.text"` which is `${id_within_lesson}|stage.${part_id}.${input_name}` ... anyway, there shouldn't be any collisions because every activity instance should have its own attempt_guid.
this is building a map that looks like:
```
{
"q:1555699921528:664|stage.Name.text": "Herman",
"q:1555699921528:664|stage.Name.textLength": 6,
}
```
that will be merged into a larger map that contains all of those inputs for each part.
the conditions of the rules are going to be looking for the keys to be in that format